libraries/ESP8266mDNS/src/LEAmDNS_Structs.cpp

```diff
@@ -1666,7 +1666,7 @@ bool MDNSResponder::stcMDNSServiceQuery::stcAnswer::stcTTL::finalTimeoutLevel(vo
 unsigned long MDNSResponder::stcMDNSServiceQuery::stcAnswer::stcTTL::timeout(void) const
 {
 
-    uint32_t    u32Timeout = esp8266::polledTimeout::oneShotMs::neverExpires;
+    uint32_t    u32Timeout = (uint32_t)esp8266::polledTimeout::oneShotMs::neverExpires;
```

Comment:
this actually seems wrong, we strip 64bit type into 32bit one.
neverExpires is `decltype(millis())` aka unsigned long (which is also a return type)
```cpp
static_assert(sizeof(unsigned long) == 8, ""); // on host
```

multiply below should also be fixed then?

Reply:
I ended up essentially there.  This is an error/invalid condition per the comments, anyway, so it's not feeling excessively critical.

Reply:
I suggest leaving it as it were (still with the warning) and fix in another PR.

Or, if you have time, it looks like this `unsigned long` / `uint32_t` should be a `decltype(esp8266::polledTimeout::oneShotMs::getTimeout())` (which is an internal `timeType` and in the end a `decltype(millis())` = `unsigned long`... without warning or suspicious casting)
see LEAmDNS.h's `esp8266::polledTimeout::oneShotMs m_TTLTimeout;`